src/test/java/org/openrewrite/java/migrate/MXBeanNonPublicTest.java

```diff
@@ -0,0 +1,304 @@
+                      server.createMBean("bean", objectName);
+                  }
+              
+                  class NonPublic implements NonPublicMBean {
+                  }
+              
+                  @MXBean
+                  private interface NonPublicMBean {
+                  }
+              }
+              """,
+            """
+              import javax.management.InstanceAlreadyExistsException;
+              import javax.management.MBeanException;
+              import javax.management.MBeanRegistrationException;
+              import javax.management.MBeanServer;
+              import javax.management.MXBean;
+              import javax.management.NotCompliantMBeanException;
+              import javax.management.ObjectName;
+              import javax.management.ReflectionException;
+                             
+              class TestClassAndNonPublicInterface {
+                  public void regMBean(ObjectName objectName, MBeanServer server) throws InstanceAlreadyExistsException, MBeanRegistrationException, NotCompliantMBeanException {
+                      Object obj = new NonPublic();
+                      server.registerMBean(obj, objectName);
+                  }
+              
+                  public void createMBean(ObjectName objectName, MBeanServer server) throws InstanceAlreadyExistsException, NotCompliantMBeanException, ReflectionException, MBeanException {
+                      server.createMBean("bean", objectName);
+                  }
+              
+                  class NonPublic implements NonPublicMBean {
+                  }
+              
+                  @MXBean
+                  public interface NonPublicMBean {
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void nonPublicInterfaceAnnotationMXBean() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import javax.management.MXBean;
+              
+              @MXBean
+              interface Foo {
+              }
+              """,
+            """
+              import javax.management.MXBean;
+              
+              @MXBean
+              public
+              interface Foo {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void nonPublicInterfaceAnnotationTrueMXBean() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              import javax.management.MXBean;
+                                
+              @MXBean(true)
+              interface Foo {
+              }
+              """,
+            """
+              import javax.management.MXBean;
+                             
+              @MXBean(true)
+              public
+              interface Foo {
```

Comment:
Not quite sure where this extra newline comes from; ideally we have that removed before a merge, but since this will likely only affect a small number of interfaces I can also see that as a future improvement.

Reply:
Haha yeah, I wasn't able to determine the cause. I was hoping you might know.

Reply:
Yeah I tried for a bit to figure out a way to fix it; but in the end thought it not worth spending considerable amount of time on, if we're not even sure it replicates beyond this test.